s3_tools/utils.py

```diff
@@ -21,3 +21,46 @@ def _get_future_output(future: futures.Future) -> Any:
         return future.result()
     except Exception as e:
         return repr(e)
+
+
+def _create_progress_bar(description: str, length: int):
+    """Create a console progress bar using 'rich' package.
+
+    Parameters
+    ----------
+    description : str
+        Progress bar description.
+    length : int
+        Progress bar length.
+
+    Returns
+    -------
+    Tuple[Optional[Progress], Optional[int]]
+        The progress bar object and the task ID associated.
+        (Need to install extra [progress] to be used)
+    """
+    try:
+        from rich.progress import (
+            BarColumn,
+            Progress,
+            TextColumn
+        )
+
+        progress = Progress(
+            TextColumn("[bold blue]{task.description}", justify="right"),
+            BarColumn(bar_width=None),
+            TextColumn("[bold blue][{task.completed}/{task.total}]"),
+        )
+
+        task_id = progress.add_task(
+            description=description,
+            total=length,
+            start=False
+        )
+
+    except ImportError:
+        print("Please install rich package. "
```

Comment:
Shouldn't it be something line: Please run `pip install aws-s3-tools[progress]` to have access to the progress bar.

Reply:
True 👍
I'll fix it, nice catch